File: src/Linked List/Remove A Node.cpp
#include <bits/stdc++.h>
using namespace std;

typedef struct node Node;
Node *head;
struct node {
    int data;
    Node *next;
};

Node *remove_node (Node *head, Node *node)
{
    if (node == head) {
        head = node->next;
        free(node);
        return head;
    }
    Node *current_node = head;
    while (current_node != NULL) {
        if (current_node->next == node) break;
        current_node = current_node->next;
    }
    if (current_node == NULL) {
        return head;
    }
    current_node->next = node->next;
    free(node);
    return head;
}

int main()
{
    Node *a = (Node *) malloc(sizeof(Node));
    Node *b = (Node *) malloc(sizeof(Node));
    Node *c = (Node *) malloc(sizeof(Node));

    a->data=10;
    a->next=b;
    b->data=20;
    b->next=c;
    c->data=30;
    c->next=NULL;
    
    head = a;
    
    remove_node(head, b);
    
    while(head!=NULL)
    {
        printf("%d\n",head->data);
        head = head->next;
    }
    
    return 0;
}

File: src/Data_Structure/Linked_List/Remove_A_Node.java
package Data_Structure.Linked_List;

public class Remove_A_Node {

  static Node head;
  static Node removeNode(Node head, Node node) {
    if (node == head) {
      head = node.next;
      return head;
    }

    Node currentNode = head;

    while (currentNode != null) {
      if (currentNode.next != node) {
        break;
      }
      currentNode = currentNode.next;
    }

    if (currentNode == null) {
      return head;
    }

    currentNode.next = node.next;

    return head;
  }

  public static void main(String[] args) {
    Node a = new Node();
    Node b = new Node();
    Node c = new Node();

    a.data = 10;
    a.next = b;

    b.data = 20;
    b.next = c;

    c.data = 30;
    c.next = null;

    head = a;

    removeNode(head, b);

    Node currentNode = head;

    while (currentNode != null) {
      System.out.printf("%d\n", head.data);
      currentNode = currentNode.next;
    }

  }
}
